### PR TITLE
libnvvpi1: add recipe

### PR DESCRIPTION
**Pull request by ichergui**
Add recipe for installing the Nvidia VPI library.

Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>

### CODE REVIEW

**Comment by ichergui** [comment]
Once this PR will be merged, I will push new one concerning the `vpi-samples` in the repo [meta-tegra-support](https://github.com/OE4T/tegra-demo-distro/tree/master/layers/meta-tegra-support)

File: recipes-devtools/vpi/libnvvpi1/0001-vpi-config-allow-to-update-the-installation-prefix.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+diff --git a/usr/share/vpi1/cmake/vpi-config.cmake b/usr/share/vpi1/cmake/vpi-config.cmake
+index 638a97d..87de6ee 100644
+--- a/usr/share/vpi1/cmake/vpi-config.cmake
++++ b/usr/share/vpi1/cmake/vpi-config.cmake
+@@ -42,7 +42,12 @@ unset(_expectedTargets)
+ 
+ 
+ # The installation prefix configured by this project.
+-set(_IMPORT_PREFIX "/opt/nvidia/vpi1")
++if (DEFINED ENV{TEGRA_SYSROOT})
```
Comment (madisongh) [inline recipes-devtools/vpi/libnvvpi1/0001-vpi-config-allow-to-update-the-installation-prefix.patch:20]: Is there a way to do this without having to introduce an extra environment variable? Some already-defined cmake variable we can use?

File: recipes-devtools/vpi/libnvvpi1_1.0.15.bb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+INHIBIT_PACKAGE_STRIP = "1"
+INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
+INHIBIT_SYSROOT_STRIP = "1"
+
+PACKAGES = "${PN} ${PN}-dev"
+FILES_${PN} = "/opt/nvidia/vpi1/lib64 ${sysconfdir}/ld.so.conf.d"
+FILES_${PN}-dev += "/opt/nvidia/vpi1/include ${datadir}/vpi1/cmake"
+PROVIDES += "${PN}-dev"
```
Comment (madisongh) [inline recipes-devtools/vpi/libnvvpi1_1.0.15.bb:50]: This PROVIDES shouldn't be needed.

Comment (ichergui) [reply recipes-devtools/vpi/libnvvpi1_1.0.15.bb:50]: I think it needed for building the VPI sample application.
I will double check.

File: recipes-devtools/vpi/libnvvpi1/0001-vpi-config-allow-to-update-the-installation-prefix.patch (hunk 1 above)
Comment (ichergui) [reply recipes-devtools/vpi/libnvvpi1/0001-vpi-config-allow-to-update-the-installation-prefix.patch:20]: I don't think so, the original code is as follow:

```
...
# The installation prefix configured by this project.
set(_IMPORT_PREFIX "/opt/nvidia/vpi1")

# Create imported target vpi
add_library(vpi SHARED IMPORTED)

set_target_properties(vpi PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)
...
```
Nvidia use a hardcoded value.

Comment (madisongh) [reply recipes-devtools/vpi/libnvvpi1/0001-vpi-config-allow-to-update-the-installation-prefix.patch:20]: > Nvidia use a hardcoded value.

Yes, but other packages that provide cmake modules don't appear to need an extra environment variable for integration into the Yocto cross-build system.

File: recipes-devtools/vpi/libnvvpi1_1.0.15.bb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+INHIBIT_PACKAGE_STRIP = "1"
+INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
+INHIBIT_SYSROOT_STRIP = "1"
+
+PACKAGES = "${PN} ${PN}-dev"
+FILES_${PN} = "/opt/nvidia/vpi1/lib64 ${sysconfdir}/ld.so.conf.d"
+FILES_${PN}-dev += "/opt/nvidia/vpi1/include ${datadir}/vpi1/cmake"
+INSANE_SKIP_${PN} = "dev-so"
```
Comment (madisongh) [inline recipes-devtools/vpi/libnvvpi1_1.0.15.bb:50]: If you fix the above, you should not need this INSANE_SKIP.

**Review by madisongh — CHANGES_REQUESTED** [review]
Mostly looks good.  Just some minor changes needed to packaging.

File: recipes-devtools/vpi/libnvvpi1_1.0.15.bb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+SYSROOT_DIRS_append = " /opt"
+
+COMPATIBLE_MACHINE = "(tegra)"
+
+do_compile() {
+    :
+}
+
+do_install() {
+    install -d ${D}/opt/nvidia/vpi1/lib64
```
Comment (madisongh) [inline recipes-devtools/vpi/libnvvpi1_1.0.15.bb:33]: I think there's a stray blank at the end of this line that should be removed.

**Review by madisongh — APPROVED** [review]

**Pull request merged**